coffee-backend/src/main/kotlin/chat/letscoffee/security/repository/UserRepository.kt

```diff
@@ -9,4 +9,5 @@ import java.util.*
 interface UserRepository : JpaRepository<User?, Long?> {
     fun findByEmail(email: String?): Optional<User>
     fun existsByEmail(email: String?): Boolean
+    fun findByNameOrEmail(name: String?, email: String?): User?
```

Comment:
Remove method and replace usage with `findByEmail`

Reply:
We might want to have a unique username, ala snapshot, so you don't have to add your friends by email.

Reply:
True, we should make it unique. But even with a unique username, the query could return two different users, leading to an exception. I suggest splitting it into `findByName` and `findByEmail`, and using the two methods based what we know we have as input from the user/API